### PR TITLE
Update jax random shuffle API

### PR DESCRIPTION
**Pull request by sachinprasadhs**
jax.random.shuffle API is deprecated and is replaced with [jax.random.permutation](https://jax.readthedocs.io/en/latest/_autosummary/jax.random.permutation.html).

Fixes: https://github.com/keras-team/keras/issues/19271

### CODE REVIEW

File: keras/backend/jax/random.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def dropout(inputs, rate, noise_shape=None, seed=None):
 
 def shuffle(x, axis=0, seed=None):
     seed = jax_draw_seed(seed)
-    return jax.random.shuffle(seed, x, axis)
+    return jax.random.permutation(seed, x, axis, independent=True)
```
Comment (fchollet) [inline keras/backend/jax/random.py:86]: Can you provide some explanation about `independent=True`? The docs weren't super helpful unfortunately https://jax.readthedocs.io/en/latest/_autosummary/jax.random.permutation.html

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras/backend/jax/random.py (hunk 1 above)
Comment (fchollet) [reply keras/backend/jax/random.py:86]: Oh I see it's what is recommend in the deprecation message.

**Review by fchollet — APPROVED** [review]
